plugin/src/main/java/org/opensearch/ml/stats/MLModelStats.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.stats;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+public class MLModelStats implements ToXContentFragment, Writeable {
+
+    /**
+     * Model stats.
+     * Key: Model Id.
+     * Value: MLActionStats which contains action stat/value map.
+     *
+     * Example: {predict: { request_count: 1}}
+     */
+    private Map<ActionName, MLActionStats> modelStats;
+
+    public MLModelStats(StreamInput in) throws IOException {
+        this.modelStats = in.readMap(stream -> stream.readEnum(ActionName.class), MLActionStats::new);
+    }
+
+    public MLModelStats(Map<ActionName, MLActionStats> modelStats) {
+        this.modelStats = modelStats;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (modelStats != null && modelStats.size() > 0) {
```

Comment:
Missed `out.writeBoolean(true)`?  Also fix `public MLModelStats(StreamInput in)` ?

Reply:
Sorry for the confusion. I didn't clearly understand your comment. This class is pretty much similar to [this class](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/stats/MLAlgoStats.java)